### DISCUSSION TITLE
Support for switching theme automatically

### DISCUSSION THREAD

**Discussion by ghost** [aristocratos/btop#356] [Ideas]
MacOS has the setting to switch the OS appearance between light or dark mode.
This can also be done automatically depending on the time of day.
I really like that feature because dark modes are often really hard to see in bright environments during daytime, but I don't want to be blinded at night either.

Is there currently any way to go along with that switch and automatically switch the theme in btop as well, based on the current "appearance" used by the OS?
Might even be nice to allow this sort of feature for the theme, so that "paired" themes could be made, one for light, and one for dark mode; for a consistent look either way.

I'm open for any ideas to make this work.

**Comment by userdocs** [comment]
Are you talking about something like this with your local terminal and the mac osx version of btop?

https://arslan.io/2021/02/15/automatic-dark-mode-for-terminal-applications/

**Reply by ghost** [reply]
Yes, this is exactly what I was talking about!
Thanks for posting that great link, I'll be using/adapting a few of those scripts.

Editing the configs through `sed` etc. was something that I didn't really want to do, because it seems like a really hacky and potentially error-prone solution, but I'll probably have to do this for certain programs anyway, that just don't support this use case.

Would still love to have proper support for it in btop, since it's such a great program.
